components/dada-parse/src/parser/expr.rs

```diff
@@ -359,22 +359,15 @@ impl CodeParser<'_, '_> {
         }
     }
 
-    /// Parses `[permission-mode] [atomic] x = expr`
+    /// Parses `let [atomic] x = expr`
     #[tracing::instrument(level = "debug", skip_all)]
     fn parse_local_variable_decl(&mut self) -> Option<Expr> {
-        // Scan ahead, looking for either `x = ` or `x :`.
-        self.testahead(|this| {
-            let _ = this.parse_atomic();
-            this.parse_name().is_some()
-                && (this.eat_op(Op::Colon).is_some() || this.eat_op(Op::Equal).is_some())
-        })?;
-
-        // Look for `[mode] x = `. If we see that, we are committed to this
-        // being a local variable declaration. Otherwise, we roll fully back.
+        let (let_span, _) = self.eat(Keyword::Let)?;
+
         let atomic = self.parse_atomic();
         let name = self.parse_name().unwrap();
```

Comment:
we need to report an error here -- e.g., people could write `let = 5`.

Reply:
For `let = 5`, does it make sense to introduce some kind of dummy empty `Name` in the ast between `let` and `=`, like for the value with `or_dummy_expr`?

This way an error will be emitted but it will be parsed. Otherwise `unexpected token` will also be emitted later for `let`.

Reply:
nevermind, I think I found a better solution, parse the whole declaration into an `Expr::Error`